PWGHF/HFC/Tasks/taskCorrelationD0.cxx

```diff
@@ -0,0 +1,509 @@
+      registry.fill(HIST("hdeclengthxy"), candidate.decayLengthXY(), candidate.pt());
+      registry.fill(HIST("hd0Prong0"), candidate.impactParameter0(), candidate.pt());
+      registry.fill(HIST("hd0Prong1"), candidate.impactParameter1(), candidate.pt());
+      registry.fill(HIST("hd0d0"), candidate.impactParameterProduct(), candidate.pt());
+      registry.fill(HIST("hCTS"), CosThetaStarD0(candidate), candidate.pt());
+      registry.fill(HIST("hCt"), CtD0(candidate), candidate.pt());
+      registry.fill(HIST("hCPA"), candidate.cpa(), candidate.pt());
+      registry.fill(HIST("hEta"), candidate.eta(), candidate.pt());
+      registry.fill(HIST("hselectionstatus"), candidate.isSelD0() + (candidate.isSelD0bar() * 2), candidate.pt());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter0(), candidate.pt());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter1(), candidate.pt());
+      registry.fill(HIST("hDecLenErr"), candidate.errorDecayLength(), candidate.pt());
+      registry.fill(HIST("hDecLenXYErr"), candidate.errorDecayLengthXY(), candidate.pt());
+    }
+  }
+
+  template <typename TTarget, typename TTracksTrig, typename TTracksAssoc>
+  void fillCorrelations(TTarget target, TTracksTrig tracks1, TTracksAssoc tracks2, float multiplicity, float posZ)
+  {
+    auto triggerWeight = 1;
+    auto associatedWeight = 1;
+
+    for (auto& track1 : tracks1) {
+
+      float eta1 = track1.eta();
+      float pt1 = track1.pt();
+      float phi1 = track1.phi();
+      o2::math_utils::bringTo02Pi(phi1);
+
+      //  TODO: add getter for NUE trigger efficiency here
+
+      //  calculating inv. mass to be filled into the container below
+      //  Note: this is needed only in case of HF-hadron correlations
+      bool fillingHFcontainer = false;
+      float invmass = 0;
+      if constexpr (std::is_same_v<hfCandidates, TTracksTrig>) {
+        fillingHFcontainer = true;
+        invmass = InvMassD0(track1);
+        //  TODO: Check how to put this into a Filter
+        if (!isAcceptedCandidate(track1)) {
+          continue;
+        }
+      }
+
+      //  fill single-track distributions
+      target->getTriggerHist()->Fill(CorrelationContainer::kCFStepReconstructed, pt1, multiplicity, posZ, triggerWeight);
+
+      for (auto& track2 : tracks2) {
+
+        if constexpr (std::is_same_v<TTracksAssoc, TTracksTrig>) { // case of h-h correlations where the two types of tracks are the same
+          if (track1 == track2) {
+            continue;
+          }
+        }
+
+        float eta2 = track2.eta();
+        float pt2 = track2.pt();
+        float phi2 = track2.phi();
+        o2::math_utils::bringTo02Pi(phi2);
+
+        //  TODO: add getter for NUE associated efficiency here
+
+        //  TODO: add pair cuts on phi*
+
+        float deltaPhi = phi1 - phi2;
+        //  set range of delta phi in (-pi/2 , 3/2*pi)
+        deltaPhi = RecoDecay::constrainAngle(deltaPhi, -0.5 * M_PI);
```

Comment:
`M_PI` is not standard C++ and is not guaranteed to be defined. Use `PI` instead.